### PR TITLE
Update master, simplify isotope scorer

### PR DESCRIPTION
**Pull request by robinschmid**
- updated to master branch
- use best isotope pattern from row
- use cosine similarity for scoring
- the isotope score calculator is a bit outdated

# TODO
- how to handle different resolutions - sum remaining signals in the ScanAlignment

### CODE REVIEW

File: src/main/java/io/github/mzmine/modules/dataprocessing/id_localcsvsearch/DatabaseIsotopeRefinerScanBased.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      var similarity = SpectralSimilarityFunction.compositeCosine.getSimilarity(Weights.SQRT, 0,
+          HandleUnmatchedSignalOptions.KEEP_ALL_AND_MATCH_TO_ZERO, mzTolerance, 0,
+          predictedIsotopes, measuredIsotopes);
+      // also match with library as ground truth to give more weight to predicted signals
+      var similarityLibrary = SpectralSimilarityFunction.compositeCosine.getSimilarity(Weights.SQRT,
+          0, HandleUnmatchedSignalOptions.KEEP_LIBRARY_SIGNALS, mzTolerance, 0, predictedIsotopes,
+          measuredIsotopes);
+
+      if (similarity != null && similarityLibrary != null) {
+        var score = (float) (similarity.getScore() + similarityLibrary.getScore() * 2) / 3f;
```
Comment (Sonja-analytik) [inline src/main/java/io/github/mzmine/modules/dataprocessing/id_localcsvsearch/DatabaseIsotopeRefinerScanBased.java:192]: why do you do this? Seems a bit random to me, wouldn't the option of choosing between one of the two scores be better?

Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_localcsvsearch/DatabaseIsotopeRefinerScanBased.java:192]: We can remove this later and finalize once you have the isotope patterns for different resolutions

Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_localcsvsearch/DatabaseIsotopeRefinerScanBased.java:192]: Was my idea on how to calculate a combined score

Comment (Sonja-analytik) [reply src/main/java/io/github/mzmine/modules/dataprocessing/id_localcsvsearch/DatabaseIsotopeRefinerScanBased.java:192]: ok 👍

**Comment by Sonja-analytik** [comment]
@robinschmid can you merge the latest master to resolve the conflicts, please?

**Comment by robinschmid** [comment]
Yes. Want to do it together to see how to resolve conflicts?

**Comment by robinschmid** [comment]
Okay there were actually no conflicts

**Comment by Sonja-analytik** [comment]
Sorry Robin, I didn't see your message until now. Thank you!

**Pull request merged**